packages/ra-core/src/preferences/PreferencesEditorContext.tsx

```diff
@@ -2,22 +2,20 @@ import * as React from 'react';
 import { createContext, ReactElement } from 'react';
 
 export const PreferencesEditorContext = createContext<
-    PreferencesEditorContextValue
+    PreferencesEditorContextValue | undefined
 >(undefined);
 
-export type PreferencesEditorContextValue =
-    | {
-          editor: ReactElement | null;
-          setEditor: React.Dispatch<React.SetStateAction<ReactElement>>;
-          preferenceKey?: string;
-          setPreferenceKey: React.Dispatch<React.SetStateAction<string>>;
-          title: string | null;
-          titleOptions?: any;
-          setTitle: (title: string, titleOptions?: any) => void;
-          isEnabled: boolean;
-          enable: () => void;
-          disable: () => void;
-          path: string | null;
-          setPath: (path: string) => void;
-      }
-    | undefined;
+export type PreferencesEditorContextValue = {
+    editor: ReactElement | null;
+    setEditor: React.Dispatch<React.SetStateAction<ReactElement | null>>;
+    preferenceKey: string | null;
```

Comment:
I wonder why you prefer null to undefined here? Is this change necessary?

Reply:
We could but it looked more coherent with the other props